pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -112,19 +111,15 @@ func (c *connectivityCheckController) Sync(ctx context.Context, syncContext fact
 		return err
 	}
 
-	crd, getCRDErr := c.apiextensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, podnetworkconnectivitychecksCRDName, metav1.GetOptions{})
-	if getCRDErr != nil && !errors.IsNotFound(getCRDErr) {
-		return getCRDErr
+	// TODO remove the following call in in 4.8
+	// re-create crd if deleted during an upgrade from 4.6
+	err = ensureConnectivityCheckCRDExists(ctx, syncContext, c.apiextensionsClient)
```

Comment:
Should this be in the `if enabled` conditional? if the controller is disabled, there is no need to ensure/create the CRD?

Reply:
> Should this be in the `if enabled` conditional? if the controller is disabled, there is no need to ensure/create the CRD?

A different copy of the controller might be enabled, and expecting the CRD to already exist (as the end goal is that the CRD is expected to be available). For example:
* 4.6 installed, all disabled.
* During 4.7 upgrade:
  * CRD installed by CNO manifests.
  * v4.6 oas-o or v4.6 kas-o deletes CRD.
  * v4.7 CNO can't fails because CRD is missing.

So, the old/new versions will fight over the CRD during a upgrade, but the end result is that the CRD is installed regardless of enabled/disabled state, just as if it was a fresh install.

Reply:
I see. That makes sense. Can you please include the steps in the comments above in the actual code as comments?